File: client/src/main/java/commands/pcommands/Exit.java
package commands.pcommands;

import commands.*;
import utils.Exitable;

public final class Exit extends AbstractCommand {
  private final Exitable target;

  public Exit(Exitable target) {
    super(CommandInfo.of("exit", "Exit without saving collection", true, 0, false, ExecutionMode.CLIENT));
    this.target = target;
  }
  
  @Override
  public ExecutionResult execute(ExecutionPayload payload) {
    target.exit();
    return ExecutionResult.valueOf(true, "Exit command executed successfully");
  }
}

File: client-ui/domain/src/commonMain/kotlin/ru.moleus.kollector/domain/commands/pcommands/Exit.kt
package ru.moleus.kollector.domain.commands.pcommands

import commands.*
import interfaces.Exitable

class Exit(private val target: Exitable) : AbstractCommand(
    CommandInfo.of(
        "exit",
        "Exit without saving collection",
        true,
        0,
        CommandType.OTHER,
        ExecutionMode.CLIENT
    )
) {
    override fun execute(payload: ExecutionPayload): ExecutionResult {
        target.exit()
        return ExecutionResult.valueOf(true, "Exit command executed successfully")
    }
}